File: Backend/src/main/java/org/example/backend/dto/TeacherSubjectDto.java
package org.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Value;

import java.io.Serializable;

/**
 * DTO for {@link org.example.backend.domain.TeacherSubject}
 */
@Data
@NoArgsConstructor
public class TeacherSubjectDto implements Serializable {
    TeacherSubjectIdDto id;
    TeacherDto teacher;
    SubjectDto subject;
}

File: Frontend/src/app/dto/teacher-subject-dto.ts
import {TeacherSubjectIdDto} from "./teacher-subject-id-dto";
import {TeacherDto} from "./teacher-dto";
import {SubjectDto} from "./subject-dto";

export interface TeacherSubjectDto {
  id : TeacherSubjectIdDto
  teacher : TeacherDto
  subject : SubjectDto
}
